onnxruntime/core/optimizer/lora_conv1d_replacement.cc

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/lora_conv1d_replacement.h"
+#include "core/graph/graph_utils.h"
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+/*
+  in LoRA code, it will use conv1d to do projection for qkv,
+  while the conv1d calculation is mathematically equivalent to matmul, and matmul is much faster than conv1d.
+  the subsitution of the graph optimizer is: 1 conv1d >> 2 split + 1 squeeze + group_num matmul + 1 concat
```

Comment:
I am not sure I can get the idea directly from the code. Can you elaborate more details with comments about how the transformation is done, for example like this: 

https://github.com/microsoft/onnxruntime/blob/21ef14476b714c3c77a2051ea220df15a8527c5a/orttraining/orttraining/core/optimizer/compute_optimizer/padding_elimination.h#L47